core/Config.php

```diff
@@ -455,4 +457,25 @@ public static function setSetting($sectionName, $name, $value)
         $section[$name] = $value;
         self::getInstance()->$sectionName = $section;
     }
-}
+
+    /**
+     * Sanity check a config file by checking contents
+     *
+     * @param string $localPath
+     * @return bool
+     */
+    public function sanityCheck(string $localPath, $expectedContent): bool
+    {
+        clearstatcache();
```

Comment:
@JasonMortonNZ be good to only invalidate the stat cache for the specific file ideally (second parameter)